sdk/core/core-rest-pipeline/test/bearerTokenAuthenticationPolicy.spec.ts

```diff
@@ -399,3 +922,162 @@ class MockRefreshAzureCredential implements TokenCredential {
     };
   }
 }
+interface Challenge {
+  testName: string;
+  challenge: string;
+  expectedResponseCode: number;
+  expectedClaims: string | null;
+}
+
+const caeTestCases: Challenge[] = [
+  {
+    testName: "unexpected error value",
+    challenge: `Bearer authorization_uri="https://login.windows.net/", error="invalid_token", claims="ey=="`,
+    expectedResponseCode: 401,
+    expectedClaims: null,
+  },
+  {
+    testName: "cannot parse claims",
+    challenge: `Bearer claims="not base64", error="insufficient_claims"`,
+    expectedResponseCode: 401,
+    expectedClaims: null,
+  },
+  {
+    testName: "more parameters, different order",
+    challenge: `Bearer realm="", authorization_uri="http://localhost", client_id="00000003-0000-0000-c000-000000000000", error="insufficient_claims", claims="ey=="`,
+    expectedResponseCode: 200,
+    expectedClaims: "{",
+  },
+  {
+    testName: "some params don't have quotes",
+    challenge: `Bearer realm="test", algorithm=SHA-256, error="insufficient_claims", claims="eyJhY2Nlc3NfdG9rZW4iOnsibmJmIjp7ImVzc2VudGlhbCI6dHJ1ZSwidmFsdWUiOiIxNzI2MDc3NTk1In0sInhtc19jYWVlcnJvciI6eyJ2YWx1ZSI6IjEwMDEyIn19fQ=="`,
```

Comment:
I believe we can reasonably leave client behavior undefined for challenges like this because CAE challenges have quoted values

Reply:
For this case, all the params relevant for CAE have quotes, so it's expected to be able to parse this and disregard the non-quotes params here

Reply:
Okay, but CAE challenges have quoted parameters. There's no real scenario in which a CAE challenge has unquoted parameters or includes unrelated parameters from other kinds of challenge. So, we don't need to make any promises about our client behavior in such scenarios

Reply:
I agree. However, this is the wrong context for such test coverage because it isn't related to CAE. We don't need to test the policy's behavior given a CAE challenge containing unquoted values, and doing so may imply to readers that this is a real scenario all clients should anticipate

Reply:
@chlowell That point makes sense to me. I did add the tests for params without quotes for both types of challenges and this is definitely not valid CAE challenge, so I will remove the test to avoid the misunderstanding.